smithy-rules-engine/src/main/resources/software/amazon/smithy/rulesengine/testutil/test-cases/substring.json

```diff
@@ -50,6 +50,36 @@
         "error": "No tests matched"
       }
     },
+    {
+      "documentation": "unicode characters always return `None`",
```

Comment:
Just a minor wording nit, ASCII is unicode. From a documentation perspective I would word this as something like "returns `None` when a string contains code points outside the first 128 (ASCII) U+0000 to U+007F"